pkg/initialdata/initialdata.go

```diff
@@ -140,15 +139,23 @@ func CheckDatastore(ctx context.Context, ds datastore) (bool, error) {
 }
 
 // InsertIntoDatastore inserts the given JSON data into datastore with write requests.
+//
+// The given data should be formatted like this: '{"collection_a": {"1": {"field_a": "foo", "field_b": "bar"}}}'
+// but the datastore.Create expected a map with FQId like this: '{"collection_a/1": {"field_a": "foo", "field_b": "bar"}}'
+// so this function also transforms the data into correct format.
 func InsertIntoDatastore(ctx context.Context, ds datastore, data []byte) error {
```

Comment:
The only thing that this function does in convert the data.

Maybe you could change the function to 
```go
func ToCreateFormat(data []byte) (creatables map[string]map[string]json.RawMessage, migrationIndex int, err error)
```

Then you can call it like this:
```go
rawData := Gettingdata(...)
createData, migrationIndex, err := ToCreateFormat(rawData)
if err != nil {...}
err := ds.Create(ctx, createData, migrationIndex)
...
```

Then you can test the ToCreateFormat function without using a Create-Mock.

Reply:
You are right but I would like to skip this step to have more time for other things. ;-)